### PR TITLE
Add the downloadIstioCtl.sh file.

### PR DESCRIPTION
**Pull request by ibm-jason**
This file is provided as a means to download
the istioctl binary.

Signed-off-by: Jason Clark <jason.clark.oss@gmail.com>

### CODE REVIEW

File: release/downloadIstioCtl.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+if [ "x${ISTIO_VERSION}" = "x" ] ; then
+  echo "Unable to get latest Istio version. Set ISTIO_VERSION env var and re-run. For example: export ISTIO_VERSION=1.0.4"
+  exit;
+fi
+
+NAME="istio-$ISTIO_VERSION"
+URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istio-${ISTIO_VERSION}-${OSEXT}.tar.gz"
+echo "Downloading $NAME from $URL ..."
+curl -L "$URL" | tar xz "$NAME/bin/istioctl"
```
Comment (esnible) [inline release/downloadIstioCtl.sh]: I was thinking of a way to create it without creating the directories.  Your choice.

```
curl -L "$URL" | tar xz --strip-components 2 "$NAME/bin/istioctl"
echo "Downloaded istioctl version $NAME .  Copy it into your path.
```

Comment (ibm-jason) [reply release/downloadIstioCtl.sh]: /cc @esnible 

Ironically I was going to ask you about stripping off the container directories, so thanks. I'll amend. 👍

**Comment by ibm-jason** [comment]
This PR addresses https://github.com/istio/istio/issues/11527.

**Comment by esnible** [comment]
The tests failed, but this cannot be due to this PR, which is adding a new file that has no test coverage!

The PR for flakey e2e-pilot-noauth-v1alpha3-v2 was closed, yet this test failed.  https://github.com/istio/istio/issues/9867  The failure here was caused by 

```
+ curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.9.2/bin/linux/amd64/kubectl
curl: (7) Failed to connect to storage.googleapis.com port 443: Connection timed out
```

The builder was literally not being able to download _kubectl_ from Google before starting the tests.

We should consider re-opening or creating a new work item for these misbehaving tests.

**Comment by ibm-jason** [comment]
@esnible Is this something I can can help fix?

**Comment by esnible** [comment]
The only required test that failed was _ci/circleci: test_.  The output is strange ... every test that ran passed, but one test just repeated

```
error	SDS grpc server for workload proxies failed to start: grpc: the server has been stopped
```

... 24743 times over the course of two seconds then gave up!

**Comment by costinm** [comment]
/hold

I know we have another download file - but telling users to run scripts from the internet on their machines is a pretty bad practice. Docs pointing to the release page seem good enough for a lot of projects.

If you really want an easy way to run istioctl on machines - we could tell them to run some "docker run ..." command - which also downloads istioctl but at least runs it in a container and doesn't have additional risks.

**Comment by ibm-jason** [comment]
/cc @costinm @esnible 

At this point, are we going with the docker container option for https://github.com/istio/istio/issues/11527, or are do we simply abandon this PR, with preference for the current method which is basically following the docs for downloading istioctl? 

If the docker container option is feasible, I can work on creating a PR for this...

**Comment by esnible** [comment]
@ibm-jason We need to figure out how to address Costin's concern.

The new file is 90% the same as the original download.  Perhaps we can add command-line only hack?  For example `curl -L https://git.io/getLatestIstio | CLI_ONLY=true sh -`.  Or we can do it with documentation telling people to do the _ getLatestIstio_ thing, even if they just want the CLI.

The Docker container idea is cool but it will be hard to get an Istio PR out of it.

**Comment by ibm-jason** [comment]
/cc @esnible 

I like the command flag idea. I'll submit a PR for that, and then we can decide on which option to take.

**Comment by linsun** [comment]
Am I reading this correctly that user needs to download the entire release archive to download istioctl based on the script?  

Can we do what kubectl does where we host the istioctl binary somewhere in our release binary url?
something like https://github.com/istio/istio/releases/download/{istio-version}/bin/linux/amd64/istioctl or daily build url from google storage?

**Comment by ibm-jason** [comment]
@linsun  Perhaps I'm mistaken but, isn't that basically what @esnible is proposing?

**Comment by esnible** [comment]
@linsun The user *thinks* `istioctl` is small but in our download it is 95% of what appears after running curl -L https://git.io/getLatestIstio .  (To verify, download 1.1 RC 3 and notice that _istioctl_ is 59MB and the whole thing is 62MB by running `du -h` in the _istio-1.1.0-rc.3_ directory.

I first proposed building _istioctl_ and having a script download just that but @costinm suggested we didn't want another script the user runs over the Internet.  Also it seemed like a lot of work to get the builds to create archives just for _istioctl_.  It would be easy to add a `ISTIOCTL_ONLY=true` option to the existing usage, so the user does `curl -L https://git.io/getLatestIstio \
   | ISTIOCTL_ONLY=true sh -`.  Under the covers it would download the whole thing, but it would throw away the 5% that is samples and helm charts.

**Comment by linsun** [comment]
@esnible @ibm-jason SG - the stats of 95% surprised me, but that is good data point.  I agree, it would be great if we could provide a convenient option as part of our install script today.

And down the road, if we support install istio off istioctl, that could be our download like linkerd has ```curl -sL https://run.linkerd.io/install | sh``` which only downloads `linkerd`.

**Comment by ibm-jason** [comment]
Not stale.

**Comment by ibm-jason** [comment]
This is a duplicate PR. Closing this one with preference for #13378.

**Pull request closed without merging**